src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeterTest.kt

```diff
@@ -158,6 +160,37 @@ class ActionMeterTest {
         )
     }
 
+    @Test
+    fun shouldHookAction() {
+        val output = CollectionActionMetricOutput(mutableListOf())
```

Comment:
If we're not reading metrics out of it, we can use `ThrowawayActionMetricOutput` and inline it

Reply:
Added it to verify if the hook doesn't affect measurement (covered with an extra assertion).

Reply:
Should hook fail the action or should it be ignored/logged in case of hook's Exceptions? WDYT?

Reply:
I think it should fail the action. If the user wants an optional hook, they can wrap it with try-catch-log themselves